types/src/network_id.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+use serde::{Deserialize, Serialize};
+
+/// A representation of the network being used in communication.
+/// There should only be one of each NetworkId used for a single node, and handshakes should verify
+/// that the NetworkId being used is the same during a handshake, to effectively ensure communication
+/// is restricted to a network.  Network should be checked that it is not the `DEFAULT_NETWORK`
+/// @TODO: Figure out why a normal struct with an enum and String doesn't work here for serialization
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct NetworkId(String);
```

Comment:
PR is coming along really well! Thought I had left a comment somewhere. I left a note in the issue. I think we want an enum here, it will make it much more flexible / usable going forward. Also I think it is okay for this to live in the config itself, was there a reason to have it types?

Reply:
Wasn't quite sure where to put it so I put it where PeerId / AccountAddress was.  I did get a message about the enum, and planning to do it next :)  The Newtype struct was a stopgap to get the tests to run and not crash on previous attempts of an enum that didn't serialize to toml.

If its only used for config/handshake purposes i think config is fine.  if we use it past just handshakes, we might want to leave it in the types to be more general

Reply:
Sweet. In general, types is intended to be our dumping ground for client interfaces. The network id is probably outside the scope of that at least at this time , /I think/.

Reply:
Sounds good, moved to config.